backends/avx/ceed-avx-blocked.c

```diff
@@ -56,6 +56,8 @@ static int CeedInit_Avx(const char *resource, Ceed ceed) {
 // Backend Register
 //------------------------------------------------------------------------------
 CEED_INTERN int CeedRegister_Avx_Blocked(void) {
-  return CeedRegister("/cpu/self/avx/blocked", CeedInit_Avx, 30);
+  const char prefix[] = "/cpu/self/avx/blocked";
+  CeedDebugEnv("Backend Register: %s", prefix);
```

Comment:
Why is this code outside `CeedRegister`?

Reply:
Because `CeedRegister` is not where the weak vs normal registration distinction is made.

Reply:
Are you saying that weak vs normal registration is also handled outside the `CeedRegister` function? Then maybe `CeedRegister` and `CeedWeakRegister`?

Reply:
`CeedInit` calls `CeedRegisterAll`
`CeedRegisterAll` calls `CeedRegister_*` for each backend
Any backend that is not compiled has a weak link instead to `CeedRegister_Weak`
`CeedRegisterAll` doesn't "know" if it is calling the actual `CeedRegister_*` or `CeedRegister_Weak`.

Reply:
I gotta agree that needing 3 lines to register a backend *and* provide debugging output is not good. We can modify `CeedRegister` to do the debug printing if requested and call `CeedRegisterImpl` for the actual work, then have `CeedRegister_Weak` do the printing and directly call `CeedRegisterImpl`. That way you get debug printing without extra work for the caller.